symengine/real_double.cpp

```diff
@@ -248,6 +248,20 @@ class EvaluateRealDouble : public EvaluateDouble<RealDouble>
             return number(std::log(std::complex<double>(d)));
         }
     }
+    virtual RCP<const Basic> floor(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<RealDouble>(x))
+        integer_class i;
+        mpz_set_d(get_mpz_t(i), std::floor(down_cast<const RealDouble &>(x).i));
```

Comment:
I think all of the integer classes can be constructed from a double.
Did you try,
```
integer_class i(std::floor(down_cast<const RealDouble &>(x).i));
```